File: lib/commonAPI/jxcore/ext/platform/iphone/impl/Jxcore.m

#import "Jxcore.h"



@implementation Jxcore



-(void) getPlatformName:(id<IMethodResult>)methodResult {
    [methodResult setResult:@"iOS"];
}

-(void) calcSumm:(int)a b:(int)b methodResult:(id<IMethodResult>)methodResult {
    [methodResult setResult:[NSNumber numberWithInt:(a+b)]];
}

-(void) joinStrings:(NSString*)a b:(NSString*)b methodResult:(id<IMethodResult>)methodResult {
    [methodResult setResult:[a stringByAppendingString:b]];
}




@end

File: lib/commonAPI/jxcore/ext/platform/android/src/com/rho/jxcore/Jxcore.java
package com.rho.jxcore;

import java.util.Map;

import com.rhomobile.rhodes.api.IMethodResult;
import com.rhomobile.rhodes.api.MethodResult;

public class Jxcore extends JxcoreBase implements IJxcore {
    
    public Jxcore(String id) {
        super(id);
    }


    @Override
    public void getPlatformName(IMethodResult result) {
          result.set("Android");	
    }


    @Override
    public void calcSumm(int a, int b, IMethodResult result) {
          result.set(a+b);
    }

    @Override
    public void joinStrings(String a, String b, IMethodResult result) {
          result.set(a+b);
    }


}